modules/tensor_mechanics/src/auxkernels/GlobalDisplacementAux.C

```diff
@@ -58,12 +60,18 @@ GlobalDisplacementAux::computeValue()
   RankTwoTensor strain;
   strain.fillFromScalarVariable(_scalar_global_strain);
 
-  const RealVectorValue & global_disp = strain * (*_current_node);
-
-  Real value = global_disp(_component);
+  for (unsigned int dir = 0; dir < _dim; ++dir)
+    for (unsigned int var = 0; var < _ndisp; ++var)
+    {
+      bool periodic = _mesh.isTranslatedPeriodic(_disp_var[var], dir);
```

Comment:
You should cache the results of this call. `isTranslatedPeriodic` performs a find that could possibly be very slow in simulations with lots of variables.